solo-chains/runtime/dancelight/src/lib.rs

```diff
@@ -3007,8 +3099,27 @@ impl tanssi_initializer::ApplyNewSession<Runtime> for OwnApplySession {
         ContainerRegistrar::initializer_on_new_session(&session_index);
 
         let invulnerables = TanssiInvulnerables::invulnerables().to_vec();
-
-        let next_collators = invulnerables;
+        let candidates_staking =
+            pallet_pooled_staking::SortedEligibleCandidates::<Runtime>::get().to_vec();
+        // Max number of collators is set in pallet_configuration
+        let target_session_index = session_index.saturating_add(1);
+        let max_collators = <CollatorConfiguration as GetHostConfiguration<u32>>::max_collators(
+            target_session_index,
+        );
+        let next_collators: Vec<_> = invulnerables
+            .iter()
+            .cloned()
+            .chain(candidates_staking.into_iter().filter_map(|elig| {
+                let cand = elig.candidate;
+                if invulnerables.contains(&cand) {
```

Comment:
Just curious, what is the possibility of this happening?

Reply:
It's as easy as looking at the list of staking candidates and adding one of them as an invulnerable. Obviously shouldn't happen intentionally, but that can be prevented by this filter_map.

Reply:
I think it would be better design if we prevent that from happening. What do you think? 

Let's leave that right now, if it is out of scope for this PR.

Reply:
We would need to prevent both marking as invulnerable an identity from staking, as well as prevent adding to staking an identity that's already an invulnerable. Would be more error prone than this.

Reply:
*disregard*